sunnypilot/modeld_v2/modeld.py

```diff
@@ -96,11 +93,11 @@ def run(self, buf: VisionBuf, wbuf: VisionBuf, transform: np.ndarray, transform_
       self.full_desire[:-1] = self.full_desire[1:]
       self.full_desire[-1] = new_desire
       self.numpy_inputs['desire'][:] = self.full_desire.reshape(self.desire_reshape_dims).max(axis=2)
-    elif self.model_runner.is_20hz and self.model_runner.is_20hz_3d:
-      self.full_desire[0, :-1] = self.full_desire[0, 1:]
-      self.full_desire[0, -1] = new_desire
-      self.numpy_inputs['desire'][:] = self.full_desire.reshape(self.desire_reshape_dims).max(axis=2)
-    elif not self.model_runner.is_20hz:
+    elif self.model_runner.is_20hz and self.model_runner.is_20hz_3d: # is_20hz and is_20hz_3d
+      self.full_desire[0,:-1] = self.full_desire[0,1:]
+      self.full_desire[0,-1] = new_desire
+      self.numpy_inputs['desire'][:] = self.full_desire.reshape((1,SplitModelConstants.INPUT_HISTORY_BUFFER_LEN,SplitModelConstants.TEMPORAL_SKIP,-1)).max(axis=2)
```

Comment:
Let's also find a way to calculate this similar to `desire_reshape_dims` as when I added it for supercombos it was because some models had different desire shape